Restructure Travis CI

AnthonyAmanse opened the pull request:
* Restructured Travis CI
* Added kubeadm for PR tests
* Added yamllint and shellcheck

mlangbehn reviewed (changes requested):
This change request does a fair bit more than just "add kubeadm for Travis CI." Let's get a more descriptive commit message on it.

AnthonyAmanse committed:
Restructure Travis CI

Previous Travis CI configuration can't test pull requests as it was dependent on Bluemix Authentication. Added kubeadm for a Kubernetes deployment in Travis CI. Added linting tests in Travis CI.

mlangbehn reviewed (approved):
Thanks!
